File: SWGradientNodeObjC/SWGradientNode_iOS/SmoothGradientScene.m
//
//  SmoothGradientScene.m
//  SWGradientNodeObjC
//
//  Created by NSSimpleApps on 03.07.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

#import "SmoothGradientScene.h"
#import "SWGradientNode.h"

@implementation SmoothGradientScene

- (void)didMoveToView:(SKView *)view {
    
    [super didMoveToView:view];
    
    self.backgroundColor = [SKColor clearColor];
    
    NSArray<SKColor *> *colors = @[[SKColor redColor],
                                   [SKColor greenColor],
                                   [SKColor blueColor],
                                   [SKColor yellowColor]];
    
    SWGradientNode *node = [[SWGradientNode alloc] initWithColors:colors
                                                        locations:nil
                                                           bounds:view.bounds
                                                     gradientType:SWGradientTypeSmooth];    
    [self addChild:node];
}

@end

File: SWGradientNodeSwift/SWGradientNode_iOS/SmoothGradientScene.swift
//
//  SmoothGradientScene.swift
//  SWGradientNodeSwift
//
//  Created by NSSimpleApps on 03.07.16.
//  Copyright © 2016 NSSimpleApps. All rights reserved.
//

import SpriteKit

class SmoothGradientScene: SKScene {
    
    override func didMoveToView(view: SKView) {
        
        super.didMoveToView(view)
        
        //self.backgroundColor = SKColor.whiteColor()
        
        let colors =
            [SKColor.redColor(),
             SKColor.greenColor(),
             SKColor.blueColor(),
             SKColor.yellowColor()
        ]
        
        let locations: [Float]? = [0.15, 0.5]//[0.15, 0.5, 0.9]
        
        let gradientNode = SWGradientNode(colors: colors, locations: locations, bounds: view.bounds, gradientType: .Smooth)
        self.addChild(gradientNode)
    }
}
